sql/src/main/java/io/crate/operation/PageResultListener.java

```diff
@@ -1,7 +1,38 @@
-package io.crate.operation;
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
 
+package io.crate.operation;
 
+/**
+ * An interface to signal that more paging data is available.
+ * The PageResultListener has to send a request to ask for more data from the source.
+ */
 public interface PageResultListener {
 
+    /**
+     * Indicates that more data can be received. The implementation has to
+     * trigger an action to request more data.
+     * See {@link io.crate.executor.transport.distributed.TransportDistributedResultAction.SendResponsePageResultListener}
```

Comment:
not sure if it's a good idea to document implementation details into an interface

Reply:
As example I think it is ok. What else than triggering something external can this method do, since it's returning void?

Reply:
throwing an exception? its all up to the implementation

Reply:
How about referencing back to https://github.com/crate/crate/pull/6681/files#diff-173f7d1183bba8fca495fad1939d503fR53 if `needMore(true)` is called, the PageBucketReceiver expects another `setBucket` calls. If `needMore=false` nothing has to be done by the listener implementation and it can free up any resources it had acquired as it won't be needed anymore